kubespawner/spawner.py

```diff
@@ -1784,6 +1851,35 @@ def _start(self):
                 else:
                     raise
 
+        if self.internal_ssl_directory:
+            secret = self.get_secret_manifest()
+            if secret:
+                try:
+                    yield self.asynchronize(
+                        self.api.create_namespaced_secret,
+                        namespace=self.namespace,
+                        body=secret
+                    )
+                except ApiException as e:
+                    if e.status == 409:
+                        self.log.info("Secret " + self.secret_name + " already exists, so did not create new secret.")
+                    else:
+                        raise
```

Comment:
When using a non highly-available k8s api-server, timeout errors will happen from time to time. Will this raise cause an error to be logged but jupyterhub keep running rather than jupyterhub crashing?

Reply:
This will cause an exception to raise and be logged, and the code will keep operating.  This is patterned after the code a few lines above.